avro/src/decimal.rs

```diff
@@ -93,7 +93,7 @@ impl Decimal {
         let mut decimal_bytes = vec![sign_byte; len];
         let raw_bytes = self.value.to_signed_bytes_be();
         let num_raw_bytes = raw_bytes.len();
-        let start_byte_index = len.checked_sub(num_raw_bytes).ok_or(Error::SignExtend {
+        let start_byte_index = len.checked_sub(num_raw_bytes).ok_or(Details::SignExtend {
```

Comment:
ok_or_else ?

Reply:
Because it's not a function call but just an enum construction it's not necessary. Clippy even wants you to use `ok_or` if you're calling .len() on something